libafl/src/feedbacks/map.rs

```diff
@@ -44,6 +47,87 @@ where
     fn reduce(first: T, second: T) -> T;
 }
 
+/// A [`MaxReducer`] reduces int values and returns their maximum.
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct AflReducer {}
+
+/* The AflReducer is very close to AFL++'s count_class_lookup8 - with
+   the difference that there:
+     [32 ... 127] = 64,
+     [128 ... 255] = 128
+   but we do
+     [32 ... 63] = 64,
+     [64 ... 255] = 128
+*/
+impl<T> Reducer<T> for AflReducer
+where
+    T: PrimInt
+        + Default
+        + Copy
+        + 'static
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + PartialOrd,
+{
+    #[inline]
+    fn reduce(_ignore: T, val: T) -> T {
+        // most are zero, second most are one, then two - handle these first
+        // these are always the same for all Integer types. speed.
+        if val <= T::from(2).unwrap() {
+            return val;
+        }
+
+        // for all other values it depends on the bit length of the PrimInt
+        // to what bit we reduce the value
+        let bit_length = size_of::<T>() as u32 * 8;
+
+        // if the value is smaller/equal half of the bitsize we dedicate
+        // one specific bit for this
+        if bit_length as u64 / 2 <= val.to_u64().unwrap() {
+            let new_val = T::one();
+            return new_val.rotate_left(val.to_u32().unwrap() - 1);
+        }
+
+        // if it is larger we keep the highest bit and shift left to compensate
+        // for the dedicated lower bits.
+        // Of course this can overflow so check for the max value first for
+        // which onwards we set the highest bit
+        let new_val = T::one();
+        match bit_length {
+            8 => {
+                if val >= T::from(64 as u8).unwrap() {
+                    new_val.rotate_left(7)
+                } else {
+                    new_val.rotate_left(bit_length - val.leading_zeros())
+                }
+            }
+            16 => {
+                if val >= T::from(2048 as u16).unwrap() {
+                    new_val.rotate_left(15)
+                } else {
+                    new_val.rotate_left(3 + bit_length - val.leading_zeros())
+                }
+            }
+            32 => {
+                if val >= T::from(1048576 as u32).unwrap() {
+                    new_val.rotate_left(31)
+                } else {
+                    new_val.rotate_left(10 + bit_length - val.leading_zeros())
+                }
+            }
+            _ => {
```

Comment:
you could specify `u64 | u128 => ` and panic for any other type or bit length (to catch bugs)

Reply:
then u64 and panic as default